buildSrc/src/main/kotlin/Utilities.kt

```diff
@@ -89,3 +92,63 @@ fun DependencyConstraint.testJarCapability() {
     )
   }
 }
+
+/** Check whether the current build is run in the context of integrations-testing. */
+fun isIntegrationsTestingEnabled() =
+  System.getProperty("nessie.integrationsTesting.enable").toBoolean()
+
+/**
+ * Adds an `implementation` dependency to `nessie-client` using the Nessie version supported by the
+ * latest released Iceberg version (`versionClientNessie`), if the system property
+ * `nessieIntegrationsTesting` is set to `true`.
+ */
+fun Project.nessieClientForIceberg(): Any {
+  if (!isIntegrationsTestingEnabled()) {
+    val clientNessieVersion = dependencyVersion("versionClientNessie")
+    return "org.projectnessie:nessie-client:$clientNessieVersion"
```

Comment:
style: a similar dependency has `dependencyVersion(...)` inside the string (e.g. on line 149) :shrug: